src/shredder/validated_shreds.rs

```diff
@@ -7,18 +7,36 @@ use crate::shredder::{TOTAL_SHREDS, ValidatedShred};
 
 /// Validated shreds array type.
 ///
-/// Using strong type to enforce certain constraints e.g. the shred indexes are valid.
+/// Using strong type to enforce certain constraints:
+/// - Shreds are in the correct order.
+/// - Shred indices match expected shred type.
+/// - Shreds are all the same size.
+#[derive(Clone, Copy)]
```

Comment:
are we sure about deriving `Copy`?  This can be a pretty expensive struct to copy.  We should be explicit about it.

Reply:
`Copy` should be fine here, because it only copies the reference